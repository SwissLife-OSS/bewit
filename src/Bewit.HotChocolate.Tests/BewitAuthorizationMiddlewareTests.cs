using System.Threading.Tasks;
using Snapshooter.Xunit;
using Xunit;

namespace Bewit.HotChocolate.Tests
{
    public class BewitAuthorizationMiddlewareTests
    {
        [Fact]
        public async Task Query_WhenAuthorize_Success()
        {
            // arrange
            var serviceProvider = TestHelpers.CreateServiceProvider();
            var payload = "foo@bar.gmail.com";
            var token = await TestHelpers.CreateToken(serviceProvider, payload);
            var schema = TestHelpers.CreateSchema(serviceProvider);

            // act
            var result = await TestHelpers.ExecuteQuery(schema, token);

            // assert
            result.MatchSnapshot();
        }

        [Fact]
        public async Task Query_WhenWrongToken_Fail()
        {
            // arrange
            var serviceProvider = TestHelpers.CreateServiceProvider();
            var token = await TestHelpers.CreateBadToken();
            var schema = TestHelpers.CreateSchema(serviceProvider);

            // act
            var result = await TestHelpers.ExecuteQuery(schema, token);

            // assert
            result.MatchSnapshot(options =>
                options.IgnoreField("Errors[0].Exception.StackTraceString"));
        }

        [Fact]
        public async Task Query_WhenNotAuthorize_Fail()
        {
            // arrange
            var serviceProvider = TestHelpers.CreateServiceProvider();
            var schema = TestHelpers.CreateSchema(serviceProvider);

            // act
            var result = await TestHelpers.ExecuteQuery(schema);

            // assert
            result.MatchSnapshot();
        }
    }
}
