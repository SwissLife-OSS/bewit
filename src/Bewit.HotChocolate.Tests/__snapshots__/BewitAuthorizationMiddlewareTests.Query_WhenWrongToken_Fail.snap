{
  "Data": {
    "foo": null
  },
  "Errors": [
    {
      "Message": "NotAuthorized",
      "Code": null,
      "Path": [
        "foo"
      ],
      "Locations": [
        {
          "Line": 1,
          "Column": 3
        }
      ],
      "Exception": {
        "ClassName": "Bewit.Validation.Exceptions.BewitInvalidException",
        "Message": "The given Bewit is invalid.",
        "Data": null,
        "InnerException": null,
        "HelpURL": null,
        "StackTraceString": "   at Bewit.Validation.BewitTokenValidator`1.ValidateBewitAsync(Bewit`1 bewit, CancellationToken cancellationToken) in /Users/michael/local/swiss/Bewit/src/Bewit.Validation/BewitTokenValidator.cs:line 80\n   at Bewit.Validation.BewitTokenValidator`1.ValidateBewitTokenAsync(BewitToken`1 bewit, CancellationToken cancellationToken) in /Users/michael/local/swiss/Bewit/src/Bewit.Validation/BewitTokenValidator.cs:line 40\n   at Bewit.HotChocolate.BewitAuthorizationMiddleware.InvokeAsync(IDirectiveContext context) in /Users/michael/local/swiss/Bewit/src/Bewit.HotChocolate/BewitAuthorizationMiddleware.cs:line 37",
        "RemoteStackTraceString": null,
        "RemoteStackIndex": 0,
        "ExceptionMethod": null,
        "HResult": -2146233088,
        "Source": "Bewit.Validation",
        "WatsonBuckets": null
      },
      "Extensions": {}
    }
  ],
  "Extensions": null,
  "ContextData": null
}
